src/sagemaker/jumpstart/cache.py

```diff
@@ -340,32 +348,34 @@ def _retrieval_function(
                 formatted_content=model_specs
             )
         if data_type == HubContentType.MODEL:
-            info = get_info_from_hub_resource_arn(
+            hub_name, _, model_name, model_version = hub_utils.get_info_from_hub_resource_arn(
                 id_info
             )
-            hub = CuratedHub(hub_name=info.hub_name, region=info.region)
-            hub_content = hub.describe_model(
-                model_name=info.hub_content_name, model_version=info.hub_content_version
+            hub_model_description: Dict[str, Any] = self._sagemaker_session.describe_hub_content(
+                hub_name=hub_name,
+                hub_content_name=model_name,
+                hub_content_version=model_version,
+                hub_content_type=data_type
             )
+
+            model_specs = JumpStartModelSpecs(DescribeHubContentsResponse(hub_model_description), is_hub_content=True)
+
             utils.emit_logs_based_on_model_specs(
-                hub_content.content_document,
+                model_specs,
                 self.get_region(),
                 self._s3_client
             )
-            model_specs = JumpStartModelSpecs(hub_content.content_document, is_hub_content=True)
             return JumpStartCachedContentValue(
                 formatted_content=model_specs
             )
-        if data_type == HubContentType.HUB:
-            info = get_info_from_hub_resource_arn(
-                id_info
-            )
-            hub = CuratedHub(hub_name=info.hub_name, region=info.region)
-            hub_info = hub.describe()
-            return JumpStartCachedContentValue(formatted_content=hub_info)
+        if data_type == HubType.HUB:
+            hub_name, _, _, _ = hub_utils.get_info_from_hub_resource_arn(id_info)
```

Comment:
can u make the return type a data class in the next PR?

Reply:
Clarifying question: Did you mean something like this? 
```
            hub_info: HubInfo = hub_utils.get_info_from_hub_resource_arn(id_info)
```

Reply:
yes, exactly!